src/bp/core/services/dialog/flow/service.ts

```diff
@@ -97,35 +108,95 @@ export class FlowService {
     }
   }
 
-  async saveAll(botId: string, flowViews: FlowView[], flowsToKeep: string[] = []) {
+  async insertFlow(botId: string, flow: FlowView) {
+    await this._upsertFlow(botId, flow)
+
+    this.notifyChanges({
+      name: flow.name,
+      modification: 'create',
+      payload: flow
+    })
+  }
+
+  async updateFlow(botId: string, flow: FlowView) {
+    await this._upsertFlow(botId, flow)
+
+    this.notifyChanges({
+      name: flow.name,
+      modification: 'update',
+      payload: flow
+    })
+  }
+
+  private async _upsertFlow(botId: string, flow: FlowView) {
     process.ASSERT_LICENSED()
-    if (!flowViews.find(f => f.name === 'main.flow.json') && flowsToKeep.indexOf('main.flow.json') === -1) {
-      throw new Error(`Expected flows list to contain 'main.flow.json'`)
+
+    const ghost = this.ghost.forBot(botId)
+
+    const flowFiles = await ghost.directoryListing(FLOW_DIR, '*.json')
```

Comment:
I know this was already like this but we actually support nested directories so the correct glob would be `**/*.json`